packages/components/Item/Item.js

```diff
@@ -89,218 +40,85 @@ const ItemComponent = ({
-  };
-
-  const handleFinishEditWorkflow = (data, { setSubmitting }) => {
-    dispatch(editItemRequest(data, setSubmitting));
-
-    notification.show({
-      text: `The ${ITEM_TEXT_TYPE[data.type]} has been updated`,
-    });
-  };
+  const handleClickAcceptEdit = ({ name, value }) => {
+    setSubmitting(true);
+    const updatedData = { ...data, listId, [name]: value };
 
-  const handleClickCancelWorkflow = () => {
-    dispatch(
-      setWorkInProgressItem(
-        item.mode === ITEM_MODE.WORKFLOW_EDIT ? item : null,
-        item.mode === ITEM_MODE.WORKFLOW_EDIT ? ITEM_MODE.REVIEW : null,
-      ),
-    );
+    dispatch(editItemRequest(updatedData, setSubmitting, notification));
   };
 
   const handleClickRestoreItem = async () => {
     dispatch(moveItemRequest(item.id, null, item.previousListId));
     dispatch(setWorkInProgressItem(null));
   };
 
-  const handleToggleFavorites = id => () => {
-    dispatch(toggleItemToFavoriteRequest(id));
-  };
-
-  const handleAcceptUpdate = id => () => {
-    dispatch(acceptItemUpdateRequest(id));
-  };
-
-  const handleRejectUpdate = id => () => {
-    dispatch(rejectItemUpdateRequest(id));
-  };
-
-  if (!item) {
-    return <EmptyItem />;
-  }
-
-  const { mode, type, update, ownerId, id } = item;
-
-  const renderedItemForm = matchStrict(
-    type,
-    {
-      [ITEM_TYPE.CREDENTIALS]: (
-        <CredentialsForm
-          item={item}
-          mode={mode}
-          onFinishCreateWorkflow={handleFinishCreateWorkflow}
-          onFinishEditWorkflow={handleFinishEditWorkflow}
-          onCancelWorkflow={handleClickCancelWorkflow}
-        />
-      ),
-      [ITEM_TYPE.DOCUMENT]: (
-        <DocumentForm
-          item={item}
-          mode={mode}
-          onFinishCreateWorkflow={handleFinishCreateWorkflow}
-          onFinishEditWorkflow={handleFinishEditWorkflow}
-          onCancelWorkflow={handleClickCancelWorkflow}
-        />
-      ),
-    },
-    null,
-  );
-
-  const access = childItems.reduce(
-    (acc, childItem) => (childItem.userId === user.id ? childItem.access : acc),
-    null,
-  );
-  const hasWriteAccess = ownerId === user.id || access === PERMISSION_WRITE;
-  const isReadOnly = access && !hasWriteAccess;
-
-  const renderUpdateNotify = () => {
-    const updateUserName = membersById[update.userId]
-      ? membersById[update.userId].name
-      : '';
+  const itemSubject = item.teamId
+    ? {
+        __typename: PERMISSION_ENTITY.TEAM_ITEM,
+        team_edit_item: !!item._links?.team_edit_item,
+        team_move_item: !!item._links?.team_move_item,
+        team_batch_share_item: !!item._links?.team_batch_share_item,
+        team_delete_item: !!item._links?.team_delete_item,
+      }
+    : {
+        __typename: PERMISSION_ENTITY.ITEM,
+        edit_item: !!item._links?.edit_item,
+        move_item: !!item._links?.move_item,
+        batch_share_item: !!item._links?.batch_share_item,
+        delete_item: !!item._links?.delete_item,
```

Comment:
Why do you use snake_case here?

Reply:
this is a backend stuff. I decided to use snake_case vars on front end for identity.